src/main/scala/com/modelfabric/sparql/stream/client/Builder.scala

```diff
@@ -67,6 +67,7 @@ object Builder {
 
       val responseMerger = builder.add(Merge[SparqlResponse](2).named("merge.sparqlResponse"))
 
+      // JC: are the broadcast and filter necessary? or it can be just a if else logic when building the flow?
```

Comment:
Yes, you are right, but still broadcast+filter seems to be the only way to do conditionals/branching. Let me know if you find a better way..  Responses are merged. My comment above was misplaced, it deals with your question as to why we need a broadcast on lines 121 and 154...